### PR TITLE
Use jsonpickle and copy on dubious references

### PR DESCRIPTION
**Pull request by chemix-lunacy**
For some reason the arg reference between instructions/model got orphaned so need to copy the array.

Otherwise switch over to the more fully featured json pickler.

### CODE REVIEW

File: src/QAT/qat/purr/compiler/builders.py [hunk 1]
```diff
@@ -65,18 +73,19 @@ def instructions(self):
         return list(self._instructions)
 
     @staticmethod
-    def deserialize(blob, model) -> "QuantumInstructionBuilder":
-        # TODO: At this point everything is a Quantum builder variation, base class losing meaning.
-        builder = QuantumInstructionBuilder(model)
-        builder._instructions = json_loads(blob, model=model)
+    def deserialize(blob) -> "QuantumInstructionBuilder":
+        builder = jsonpickle.decode(blob, context=CyclicRefUnpickler())
+        if isinstance(builder, str):
```
Comment (helmaazouz) [inline src/QAT/qat/purr/compiler/builders.py:78]: this is differently handled in https://github.com/oqc-community/qat/pull/79/files#diff-57c60dad41f425dcf0b9f66368aa616df03bfab33465678710f1b2ff82b7700a

File: src/QAT/qat/purr/compiler/instructions.py [hunk 2]
```diff
@@ -40,13 +41,7 @@ def __repr__(self):
...
 
 
-class Instruction:
-    @staticmethod
-    def deserialize(blob, model: QuantumHardwareModel):
-        return json_loads(blob, model=model)
-
-    def serialize(self):
-        return json_dumps(self)
+class Instruction: ...
```
Comment (helmaazouz) [inline src/QAT/qat/purr/compiler/instructions.py:44]: minor: a new line before the `...` ^^

**Review by helmaazouz — CHANGES_REQUESTED** [review]

File: src/QAT/qat/purr/compiler/instructions.py (hunk 2 above)
Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/compiler/instructions.py]: @hamidelmaazouz This is how black likes it. I will make #79 formatted the same.

Comment (hamidelmaazouz) [reply src/QAT/qat/purr/compiler/instructions.py:44]: then it's all fine :+1:

Comment (hamidelmaazouz) [reply src/QAT/qat/purr/compiler/instructions.py:44]: thank you

**Review by hamidelmaazouz — APPROVED** [review]

**Review by helmaazouz — APPROVED** [review]

**Pull request merged**